src/crepe/api/ParticleEmitter.h

```diff
@@ -5,38 +5,87 @@
 
 #include "Component.h"
 #include "Particle.h"
+#include "Sprite.h"
+#include "Vector2.h"
+
+// class Sprite;
 
 namespace crepe {
 
+/**
+ * \brief Data holder for particle emission parameters.
+ *
+ * The ParticleEmitter class stores configuration data for particle properties,
+ * defining the characteristics and boundaries of particle emissions.
+ */
+
 class ParticleEmitter : public Component {
 public:
-	ParticleEmitter(game_object_id_t id, uint32_t max_particles,
-					uint32_t emission_rate, uint32_t speed,
-					uint32_t speed_offset, uint32_t angle, uint32_t angleOffset,
-					float begin_lifespan, float end_lifespan);
+	/**
+	 * \brief Defines the boundary within which particles are constrained.
+	 *
+	 * This structure specifies the boundary's size and offset, as well as the
+	 * behavior of particles upon reaching the boundary limits.
+	 */
+	struct Boundary {
+		//! boundary width (midpoint is emitter location)
+		double width = 0.0;
+		//! boundary height (midpoint is emitter location)
+		double height = 0.0;
+		//! boundary offset from particle emitter location
+		Vector2 offset;
+		//! reset on exit or stop velocity and set max postion
+		bool reset_on_exit = false;
+	};
+
+	/**
+	 * \brief Holds parameters that control particle emission.
+	 *
+	 * Contains settings for the emitter’s position, particle speed, angle, lifespan,
+	 * boundary, and the sprite used for rendering particles.
+	 */
+	struct Data {
+		//! position of the emitter
+		Vector2 position;
+		//! maximum number of particles
+		const uint32_t max_particles = 0;
```

Comment:
specifieke reden voor de uint32_t in plaats van int?

Reply:
omdat je geen negatieve particles mag hebben

Reply:
kan je dan niet unsigned int gebruiken?

Reply:
Ik ben het hier eens met Wouter. Volgende de code standaard moet er een heel goede reden zijn voor het gebruik van specifieke datatypen.

Reply:
In mijn eerdere bericht staat dat het een unsigned int gaat worden. Ik ga het aanpassen.